core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -232,6 +232,18 @@ private[spark] object Utils extends Logging {
     // scalastyle:on classforname
   }
 
+  def getQuantilesValue(
+    values: IndexedSeq[Double],
+    quantiles: Array[Double]): IndexedSeq[Double] = {
+    val count = values.size
+    val indices = quantiles.map { q => math.min((q * count).toLong, count - 1) }
+    if (count > 0) {
+      indices.map(i => values(i.toInt)).toIndexedSeq
+    } else {
+      quantiles.map(_ => 0.0)
```

Comment:
Just return a Seq of the right length filled with 0.0?

Reply:
Thanks for the suggestion, addressed it.